core-common/src/main/java/org/apache/kylin/common/KylinConfigBase.java

```diff
@@ -1638,6 +1638,10 @@ public boolean isUseLocalClasspathEnabled() {
         return Boolean.parseBoolean(getOptional("kylin.engine.mr.use-local-classpath", TRUE));
     }
 
+    public int getTopNExtraSortRate() {
+        return Integer.parseInt(getOptional("kylin.engine.mr.topn-extra-sort-rate", "10"));
```

Comment:
IMO, set to 1 as default value and use the original way to merge, if user need to reduce count of sort, can set this property to a proper value by himself. Right?